src/sources/envoy_als/grpc.rs

```diff
@@ -33,68 +37,99 @@ impl AccessLogService for Service {
             tokio::select! {
                 _ = &mut shutdown => break,
                 stream_msg = in_stream.message() => {
-                    match stream_msg {
-                        Ok(omsg) => {
-                            match omsg {
-                                Some(msg) => {
-                                    let mut events = vec![];
+                    match self.handle_message(stream_msg, &mut stream_identifier).await {
+                        Ok(_) => {},
+                        Err(err) => {
+                            match err {
+                                StreamError::StreamClosed => break,
+                                StreamError::Grpc(status) => {
+                                    return Err(status);
+                                }
+                            }
+                        }
+                    }
+                },
+            }
+        }
 
-                                    if let Some(logs_entries) = msg.log_entries {
-                                        match logs_entries {
-                                            LogEntries::HttpLogs(http_logs) => {
-                                                for l in http_logs.log_entry {
-                                                    let mut evt = LogEvent::default();
-                                                    evt.insert("http_log", Value::from(l));
+        Ok(Response::new(StreamAccessLogsResponse {}))
+    }
+}
 
-                                                    if let Some(identifier) = msg.identifier.clone() {
-                                                        stream_identifier = Some(identifier);
-                                                    }
-                                                    if let Some(stream_id) = stream_identifier.clone() {
-                                                        evt.insert("identifier", Value::from(stream_id));
-                                                    }
+impl Service {
+    async fn handle_message(
+        &self,
+        stream_msg: Result<Option<StreamAccessLogsMessage>, Status>,
+        stream_identifier: &mut Option<Identifier>,
+    ) -> Result<(), StreamError> {
+        match stream_msg {
+            Ok(omsg) => match omsg {
+                Some(msg) => {
+                    self.process_logs(msg, stream_identifier).await?;
+                }
+                None => {
+                    debug!("Sender closed stream");
+                    return Err(StreamError::StreamClosed);
```

Comment:
Continuation of the above comment about whether this is an error condition or not- if it _is_ an Error condition, we can remove the debug log because an error will be logged from the internal event emission.

Reply:
This is not an error condition. It just means Envoy closed the stream to Vector